tests/test_client_cursor_async.py

```diff
@@ -433,11 +433,13 @@ async def test_rownumber(aconn):
     await cur.fetchmany(10)
     assert cur.rownumber == 12
     rns: List[int] = []
-    async for i in cur:
+    itcur = cur.__aiter__()
+    async for i in itcur:
         assert cur.rownumber
         rns.append(cur.rownumber)
         if len(rns) >= 3:
             break
+    await itcur.aclose()
```

Comment:
Why is this pattern needed? It doesn't seem a very representative way to iterate on an object.

Reply:
This is to resolve a `ResourceWarning` raised by trio; see the message of commit 88f85700b3a95dee89f8a15947fc067e9605a460.